crates/shared/src/http_solver/model.rs

```diff
@@ -185,19 +185,33 @@ impl Interaction for InteractionData {
     }
 }
 
+#[serde_as]
 #[derive(Copy, Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
 pub enum Score {
-    /// The score used for ranking.
-    #[serde(with = "u256_decimal")]
-    Score(U256),
+    /// The score value is provided as is from solver.
+    /// Success probability is not incorporated into this value.
+    Solver {
+        #[serde(with = "u256_decimal")]
+        score: U256,
+    },
     /// This option is used to indicate that the solver did not provide a score.
     /// Instead, the score should be computed by the protocol.
     /// To have more flexibility, the protocol score can be tweaked by the
     /// solver by providing a discount.
-    #[serde(with = "u256_decimal")]
-    #[serde(rename = "scoreDiscount")]
-    Discount(U256),
+    Discount {
+        #[serde(with = "u256_decimal")]
+        #[serde(rename = "scoreDiscount")]
```

Comment:
Is there a reason that `scoreDiscount` is the only camelCase field on the entire `result.json` object?

Reply:
No specific reason.